From c7464c8f0fb2b0a9d62a72fbabea2430957ae95c Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:49 -0400
Subject: [ethernet] i40e: Check PCI_IOV config to avoid compile error

Message-id: <1409658042-22689-109-git-send-email-sassmann@redhat.com>
Patchwork-id: 90303
O-Subject: [RHEL7.1 PATCH 108/281] i40e: Check PCI_IOV config to avoid compile error
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Greg Rose <gregory.v.rose@intel.com>

The call to i40e_alloc_vfs needs to be wrapped in CONFIG_PCI_IOV because
the function itself is wrapped in the same conditional compile block.

Change-ID: I663c5f1b85e5cfba0b36da8966f7db1a034f408b
Signed-off-by: Greg Rose <gregory.v.rose@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit df805f62d16db15fdb9278bcb56ee3863deebfb8)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index c964af9..1819f26 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -8368,6 +8368,7 @@ static int i40e_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
   }
  }
 
+#ifdef CONFIG_PCI_IOV
  /* prep for VF support */
  if ((pf->flags & I40E_FLAG_SRIOV_ENABLED) &&
      (pf->flags & I40E_FLAG_MSIX_ENABLED) &&
@@ -8390,6 +8391,7 @@ static int i40e_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
       err);
   }
  }
+#endif /* CONFIG_PCI_IOV */
 
  pfs_found++;
 
-- 
1.7.1